From 8e388c3d1c0f5b78745453daaf3d914c0c5406f2 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Tue, 7 May 2019 17:38:27 +0200
Subject: [PATCH 281/386] arm64: dts: cn9130: fix the pcie memory size

The PCIe MEM space size for CP0 pcie0 is bigger than for other PCIe
interfaces - reflect it in device-tree description.

Change-Id: Ic80646b80a7ea3bd14061a68c7f438ea57ff193a
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/10472
Tested-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-70x0.dtsi  |  2 +-
 arch/arm64/boot/dts/marvell/armada-80x0.dtsi  |  4 +--
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi | 25 ++++++++++---------
 arch/arm64/boot/dts/marvell/cn9130.dtsi       |  2 +-
 arch/arm64/boot/dts/marvell/cn9131-db.dtsi    |  2 ++
 arch/arm64/boot/dts/marvell/cn9132-db.dtsi    |  2 ++
 6 files changed, 21 insertions(+), 16 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-70x0.dtsi b/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
index 3e69d6cebbad..cca2a6a4cfe1 100644
--- a/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-70x0.dtsi
@@ -10,7 +10,7 @@
  */
 #define CP110_NUM				0
 #define CP110_BASE				0xf2000000
-#define CP110_PCIE_MEM_SIZE			0xf00000
+#define CP110_PCIE_MEM_SIZE(iface)		0xf00000
 #define CP110_PCIEx_CPU_IO_BASE(iface)		(0xf9000000 + (iface) *  0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)		(0xf6000000 + (iface) *  0x1000000)
 #define CP110_PCIEx_BUS_IO_BASE(iface)		(CP110_PCIEx_CPU_IO_BASE(iface))
diff --git a/arch/arm64/boot/dts/marvell/armada-80x0.dtsi b/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
index 02020f71b489..575e324f2648 100644
--- a/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-80x0.dtsi
@@ -10,7 +10,7 @@
  */
 #define CP110_NUM				0
 #define CP110_BASE				0xf2000000
-#define CP110_PCIE_MEM_SIZE			0xf00000
+#define CP110_PCIE_MEM_SIZE(iface)		0xf00000
 #define CP110_PCIEx_CPU_IO_BASE(iface)		(0xf9000000 + (iface) *  0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)		(0xf6000000 + (iface) *  0x1000000)
 #define CP110_PCIEx_BUS_IO_BASE(iface)		(CP110_PCIEx_CPU_IO_BASE(iface))
@@ -33,7 +33,7 @@
  */
 #define CP110_NUM				1
 #define CP110_BASE				0xf4000000
-#define CP110_PCIE_MEM_SIZE			0xf00000
+#define CP110_PCIE_MEM_SIZE(iface)		0xf00000
 #define CP110_PCIEx_CPU_IO_BASE(iface)		(0xfd000000 + (iface) *  0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)		(0xfa000000 + (iface) *  0x1000000)
 #define CP110_PCIEx_BUS_IO_BASE(iface)		(CP110_PCIEx_CPU_IO_BASE(iface))
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index c63b805f10c6..b25bedfd81b5 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -9,14 +9,15 @@
 
 #include "armada-common.dtsi"
 
-#define U64_TO_U32_H(addr)			(((addr) >> 32) & 0xffffffff)
-#define U64_TO_U32_L(addr)			((addr) & 0xffffffff)
+#define U64_TO_U32_H(addr)		(((addr) >> 32) & 0xffffffff)
+#define U64_TO_U32_L(addr)		((addr) & 0xffffffff)
 
-#define CP110_PCIEx_REG0_BASE(iface)		(CP110_BASE + 0x600000 + (iface) * 0x20000)
-#define CP110_PCIEx_REG1_BASE(iface)		(CP110_PCIEx_CPU_MEM_BASE(iface) + CP110_PCIE_MEM_SIZE)
+#define CP110_PCIEx_REG0_BASE(iface)	(CP110_BASE + 0x600000 + (iface) * 0x20000)
+#define CP110_PCIEx_REG1_BASE(iface)	(CP110_PCIEx_CPU_MEM_BASE(iface) + \
+					 CP110_PCIE_MEM_SIZE(iface))
 
-#define CP110_EIP197_INDEX			CP110_NUM
-#define CP110_SPI_BUS_ID(n)			((CP110_NUM * 2) + (n))
+#define CP110_EIP197_INDEX		CP110_NUM
+#define CP110_SPI_BUS_ID(n)		((CP110_NUM * 2) + (n))
 
 / {
 	/*
@@ -574,8 +575,8 @@
 			/* non-prefetchable memory */
 			CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(0))
 			U64_TO_U32_L(CP110_PCIEx_BUS_MEM_BASE(0)) U64_TO_U32_H(CP110_PCIEx_CPU_MEM_BASE(0))
-			U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(0)) U64_TO_U32_H(CP110_PCIE_MEM_SIZE)
-			U64_TO_U32_L(CP110_PCIE_MEM_SIZE)>;
+			U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(0))
+			U64_TO_U32_H(CP110_PCIE_MEM_SIZE(0)) U64_TO_U32_L(CP110_PCIE_MEM_SIZE(0))>;
 		interrupt-map-mask = <0 0 0 0>;
 		interrupt-map = <0 0 0 0 &CP110_LABEL(icu) ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
 		interrupts = <ICU_GRP_NSR 22 IRQ_TYPE_LEVEL_HIGH>;
@@ -607,8 +608,8 @@
 			/* non-prefetchable memory */
 			CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(1))
 			U64_TO_U32_L(CP110_PCIEx_BUS_MEM_BASE(1)) U64_TO_U32_H(CP110_PCIEx_CPU_MEM_BASE(1))
-			U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(1)) U64_TO_U32_H(CP110_PCIE_MEM_SIZE)
-			U64_TO_U32_L(CP110_PCIE_MEM_SIZE)>;
+			U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(1))
+			U64_TO_U32_H(CP110_PCIE_MEM_SIZE(1)) U64_TO_U32_L(CP110_PCIE_MEM_SIZE(1))>;
 		interrupt-map-mask = <0 0 0 0>;
 		interrupt-map = <0 0 0 0 &CP110_LABEL(icu) ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
 		interrupts = <ICU_GRP_NSR 24 IRQ_TYPE_LEVEL_HIGH>;
@@ -640,8 +641,8 @@
 			/* non-prefetchable memory */
 			CP110_PCIE_BUS_MEM_CFG U64_TO_U32_H(CP110_PCIEx_BUS_MEM_BASE(2))
 			U64_TO_U32_L(CP110_PCIEx_BUS_MEM_BASE(2)) U64_TO_U32_H(CP110_PCIEx_CPU_MEM_BASE(2))
-			U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(2)) U64_TO_U32_H(CP110_PCIE_MEM_SIZE)
-			U64_TO_U32_L(CP110_PCIE_MEM_SIZE)>;
+			U64_TO_U32_L(CP110_PCIEx_CPU_MEM_BASE(2))
+			U64_TO_U32_H(CP110_PCIE_MEM_SIZE(2)) U64_TO_U32_L(CP110_PCIE_MEM_SIZE(2))>;
 		interrupt-map-mask = <0 0 0 0>;
 		interrupt-map = <0 0 0 0 &CP110_LABEL(icu) ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
 		interrupts = <ICU_GRP_NSR 23 IRQ_TYPE_LEVEL_HIGH>;
diff --git a/arch/arm64/boot/dts/marvell/cn9130.dtsi b/arch/arm64/boot/dts/marvell/cn9130.dtsi
index bdd215df7c95..bbcbd04ca6a6 100644
--- a/arch/arm64/boot/dts/marvell/cn9130.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130.dtsi
@@ -27,7 +27,7 @@
 #define CP110_BASE			(CP110_BASE_OFFSET + \
 						(CP110_NUM * CP110_SPACE_SIZE))
 
-#define CP110_PCIE_MEM_SIZE		(0xf00000)
+#define CP110_PCIE_MEM_SIZE(iface)	((iface == 0) ? 0x1ff00000 : 0xf00000)
 #define CP110_PCIE_BUS_MEM_CFG		(0x82000000)
 
 /* CP110-0 Settings */
diff --git a/arch/arm64/boot/dts/marvell/cn9131-db.dtsi b/arch/arm64/boot/dts/marvell/cn9131-db.dtsi
index f4636f78e946..ee18a493e622 100644
--- a/arch/arm64/boot/dts/marvell/cn9131-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9131-db.dtsi
@@ -6,6 +6,7 @@
  */
 
 #undef CP110_NUM
+#undef CP110_PCIE_MEM_SIZE
 #undef CP110_PCIEx_CPU_IO_BASE
 #undef CP110_PCIEx_CPU_MEM_BASE
 #undef CP110_PCIEx_BUS_IO_BASE
@@ -13,6 +14,7 @@
 
 /* CP110-1 Settings */
 #define CP110_NUM			1
+#define CP110_PCIE_MEM_SIZE(iface)	(0xf00000)
 #define CP110_PCIEx_CPU_IO_BASE(iface)	(0xf8030000 + (iface) * 0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)	(0xe2000000 + (iface) * 0x1000000)
 #define CP110_PCIEx_BUS_IO_BASE(iface)	(CP110_PCIEx_CPU_IO_BASE(iface))
diff --git a/arch/arm64/boot/dts/marvell/cn9132-db.dtsi b/arch/arm64/boot/dts/marvell/cn9132-db.dtsi
index eb1ac000208e..6e1aa747a9a5 100644
--- a/arch/arm64/boot/dts/marvell/cn9132-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9132-db.dtsi
@@ -6,6 +6,7 @@
  */
 
 #undef CP110_NUM
+#undef CP110_PCIE_MEM_SIZE
 #undef CP110_PCIEx_CPU_IO_BASE
 #undef CP110_PCIEx_CPU_MEM_BASE
 #undef CP110_PCIEx_BUS_IO_BASE
@@ -13,6 +14,7 @@
 
 /* CP110-1 Settings */
 #define CP110_NUM			2
+#define CP110_PCIE_MEM_SIZE(iface)	(0xf00000)
 #define CP110_PCIEx_CPU_IO_BASE(iface)	(0xf8060000 + (iface) * 0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)	(0xe5000000 + (iface) * 0x1000000)
 #define CP110_PCIEx_BUS_IO_BASE(iface)	(CP110_PCIEx_CPU_IO_BASE(iface))
-- 
2.17.1

